src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Person } from '../../types/Person';
+
+interface Props {
+  people: Person[];
+  saveQuery: (query: string) => void;
+  setSelectedPerson: (person: Person | null) => void;
+  setFocus: React.Dispatch<React.SetStateAction<boolean>>;
+  focus: boolean;
+}
+
+export const Autocomplete: React.FC<Props> = React.memo(
+  ({ people, saveQuery, setSelectedPerson, setFocus, focus }) => {
+    const [query, setQuery] = useState('');
+
+    return (
+      <div
+        className={classNames('dropdown', { 'is-active': focus })}
+        onFocus={() => setTimeout(() => setFocus(true), 100)}
```

Comment:
Using `setTimeout` in the `onFocus` handler can lead to unexpected behavior, especially if the component re-renders before the timeout completes. Consider using a more reliable method to handle focus state changes.